### PR TITLE
[Maintenance] refactor user validation into middleware

### PR DESCRIPTION
**Pull request by tcrasset**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
   "scripts": {
     "start": "node app",
     "lint": "eslint . --max-warnings 0",
+    "lint-fix": "eslint . --fix",
```
Comment (matt-fidd) [inline package.json:10]: Could we match the syntax in actual here?

```suggestion
    "lint:fix": "eslint . --fix",
```

File: src/app-gocardless/app-gocardless.js [hunk 2]
```diff
@@ -20,13 +20,7 @@ app.get('/link', function (req, res) {
...
 export { app as handlers };
 app.use(express.json());
-app.use(async (req, res, next) => {
-  let user = await validateUser(req, res);
-  if (!user) {
-    return;
-  }
-  next();
-});
+app.use(errorMiddleware);
```
Comment (matt-fidd) [inline src/app-gocardless/app-gocardless.js:23]: should this also use `validateUserMiddleware`?

**Review by matt-fidd — CHANGES_REQUESTED** [review]
Thanks for this! A couple of minor comments

File: src/app-gocardless/app-gocardless.js (hunk 2 above)
Comment (tcrasset) [reply src/app-gocardless/app-gocardless.js:23]: Oops, imported the wrong middleware

**Comment by tcrasset** [comment]
Should be good now. Thanks for the quick review.

**Review by matt-fidd — APPROVED** [review]

**Pull request merged**